lib/hexpm/accounts/audit_log.ex

```diff
@@ -183,4 +183,16 @@ defmodule Hexpm.Accounts.AuditLog do
   defp fields(%UserHandles{}), do: [:github, :twitter, :freenode]
 
   defp multi_key(action), do: :"log.#{action}"
+
+  def all_by(%Hexpm.Repository.Package{} = package) do
+    from(l in AuditLog, where: fragment("? @> ?", l.params, ^%{package: %{id: package.id}}))
```

Comment:
I think it might be better to do the query like this, you might need to tweak it to get the casting to/from jsonb correct:

```elixir
from(l in AuditLog, where: fragment("?->'package'->'id' == ?", l.params, ^package.id))
``` 

This way it should be easier to add an index expression for it. And please add the index :)

Reply:
The query was updated in https://github.com/hexpm/hexpm/pull/817/commits/32ef325ca70ce4a1f4208eeac6f9d6334ff4837e

The index was added in https://github.com/hexpm/hexpm/pull/817/commits/a0a1994887ae6d6882d95b70dced97aa0a594a05

Reply:
Great!